docs/index.html

```diff
@@ -15,7 +17,7 @@
       var redir    = href.replace(/\/index\.(php|x?html?)$/, '/'+newPath);
       if (protocol == 'file:') {
         location.replace(redir);
-      } else if (Array('localhost', '127.0.0.1').indexOf(hostname) > -1) {
+      } else if (hostname.match(/^(localhost|192\.168\.|127\.)(\d+\.)*/)) {
```

Comment:
just to ensure: `hostname` additionally matches, for example, "localhost1.2.3." and "127.9.9.", is it intended?

Reply:
> `hostname` additionally matches, for example, "localhost1.2.3." and "127.9.9.", is it intended?

That was an oversight. Something like this might be an improvement:

    /^(localhost(?!\d)|(192\.168\.|127\.)(\d+\.))*/

E.g.,

~~~js
const redirects = [
    '127.0.0.1',
    'localhost1',
    'localhost1.2.3.',
    'localhost:80',
    'localhost:',
    'localhost'
].reduce((res, host) => {
    const m = host.match(/^(localhost(?!\d)|(192\.168\.|127\.)(\d+\.))*/);
    res[host] = m[0] ? '-> /index-local.html' : '';
    return res;
}, {});

console.table(redirects);
~~~

      ┌─────────────────┬────────────────────────┐
      │     (index)     │         Values         │
      ├─────────────────┼────────────────────────┤
      │    127.0.0.1    │ '-> /index-local.html' │
      │   localhost1    │           ''           │
      │ localhost1.2.3. │           ''           │
      │  localhost:80   │ '-> /index-local.html' │
      │   localhost:    │ '-> /index-local.html' │
      │    localhost    │ '-> /index-local.html' │
      └─────────────────┴────────────────────────┘


Then again, a user's HOSTS file could possibly have a mapping like `localhost1` in it, so I don't see a problem with being a little more inclusive.